### PR TITLE
feat: plat-5187 Add container support to LambdaWorker construct

### PR DESCRIPTION
**Pull request by stephenrob**
Adds support to LambdaWorker for specifying a container image tag from an existing ECR Repository.

Adds `container-lambda-example` to show creating the ECR Repository, Building the docker image and then creating a basic lambda worker.

### CODE REVIEW

File: lib/lambda-worker/lambda-worker.ts [hunk 1]
```diff
@@ -45,6 +47,12 @@ export class LambdaWorker extends cdk.Construct {
       );
     }
 
+    if (!this.isContainerLambda(props) && !this.isFunctionLambda(props)) {
+      throw new Error(
+        `Invalid lambdaProps only dockerImageTag/ecrRepositoryArn/ecrRepositoryName or handler/entry can be specified.`
+      );
+    }
```
Comment (mrliptontea) [inline lib/lambda-worker/lambda-worker.ts:54]: Hmm I think this check would throw when neither is specified, which I think we need such check. But another check is needed to ensure we don't have both set.

File: lib/lambda-worker/lambda-worker.ts [hunk 2]
```diff
@@ -219,4 +206,89 @@ export class LambdaWorker extends cdk.Construct {
...
+        environment: props.lambdaProps.environment,
+        memorySize: props.lambdaProps.memorySize,
+        reservedConcurrentExecutions:
+          props.lambdaProps.reservedConcurrentExecutions,
+        retryAttempts: props.lambdaProps.retryAttempts,
+        securityGroup: props.lambdaProps.securityGroup,
+        timeout: props.lambdaProps.timeout,
+        vpc: props.lambdaProps.vpc,
+        vpcSubnets: props.lambdaProps.vpcSubnets,
+      });
```
Comment (mrliptontea) [inline lib/lambda-worker/lambda-worker.ts:268]: Could these be moved to inside factory functions, `return this.createContainerLambda(props);` and `return this.createFunctionLambda(props)`, in order to make `createLambdaFunction` shorter?

File: examples/container-lambda-worker/README.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+The `cdk.json` file tells the CDK Toolkit how to execute your app.
+
+## Useful commands
+
+- `npm run build` compile typescript to js
+- `npm run watch` watch for changes and compile
+- `npm run test` perform the jest unit tests
+- `cdk deploy` deploy this stack to your default AWS account/region
+- `cdk diff` compare deployed stack with current state
+- `cdk synth` emits the synthesized CloudFormation template
```
Comment (malcyL) [inline examples/container-lambda-worker/README.md:14]: We should give some details here. And a diagram. See https://github.com/talis/talis-cdk-constructs/tree/main/examples/simple-lambda-worker

File: lib/lambda-worker/lambda-worker.ts (hunk 1 above)
Comment (stephenrob) [reply lib/lambda-worker/lambda-worker.ts:54]: There is a test case for all being set and ensuring that the error is thrown, see the array/table of tests that are ran for these options.

Comment (mrliptontea) [reply lib/lambda-worker/lambda-worker.ts:54]: Ah sorry, I commented before looking at `isContainerLambda` and `isFunctionLambda` - all good here.

**Review by malcyL — APPROVED** [review]

**Pull request merged**